napari/_qt/qt_viewer.py

```diff
@@ -233,6 +233,9 @@ def __init__(
         self.viewer._layer_slicer.events.ready.connect(self._on_slice_ready)
 
         self._on_active_change()
+        self.viewer.layers.events.inserted.connect(self._update_camera_depth)
+        self.viewer.layers.events.removed.connect(self._update_camera_depth)
+        self.viewer.dims.events.ndisplay.connect(self._update_camera_depth)
```

Comment:
This won't work when a layer's extent changes inplace... I think it's quite likely with annotations. Probably not a big deal most of the time, as there will be another similarly sized layer behind it, but still.

Maybe you should add an event for and connect to `LayerList._extent_world`?

Reply:
I noted that above but as I started to look into it I realised that it would mean invalidating and triggering layer extent change events non stop, which we built like this whole extent cache mechanism to avoid. So I would rather fix that in a new PR than this one.